### PR TITLE
Add command to create genesis accounts

### PR DESCRIPTION
**Pull request by jackcmay**
#### Problem

No support for multiple genesis accounts.  Allowing users to create genesis accounts gives everyone the ability to run their own libra mint.

#### Summary of Changes

- Move loader program `InvokeMain` now takes two commands, `CreateGenesis` and `RunProgram`

Fixes #

### CODE REVIEW

File: programs/move_loader_api/src/processor.rs [hunk 1]
```diff
@@ -44,64 +45,37 @@ pub fn process_instruction(
...
-pub struct InvokeInfo {
-    /// Sender of the "transaction", the "sender" who is calling this program
-    pub sender_address: AccountAddress,
-    /// Name of the function to call
-    pub function_name: String,
-    /// Arguments to pass to the program being invoked
-    pub args: Vec<TransactionArgument>,
+/// Command to invoke
+#[derive(Debug, Serialize, Deserialize)]
+pub enum InvokeCommand {
```
Comment (jackcmay) [inline programs/move_loader_api/src/processor.rs]: @garious Thoughts on this?  I think ideally we would want to have a new top-level instruction for creating the genesis account, on the same level as `LoaderInstruction`.

Comment (garious) [reply programs/move_loader_api/src/processor.rs]: Looks like it's getting a little weird for all the loaders to implement only `LoaderInstruction`. Instead, they should each have their own enum and implement a shared trait. The runtime can use the trait to create `InvokeMain` instructions. Beyond that, each program would be free to implement whatever additional instructions make sense for it.

Comment (jackcmay) [reply programs/move_loader_api/src/processor.rs]: Yup, agreed.  @greg I'm planning to merge this when it passes CI and address the larger `LoaderInstruction` changes in a separate PR.  Any objections?

**Comment by jackcmay** [comment]
@sakridge Any feedback on these changes?

File: programs/librapay_api/src/lib.rs [hunk 2]
```diff
@@ -18,11 +18,34 @@ use solana_runtime::loader_utils::load_program;
...
 use types::account_address::AccountAddress;
 
+pub fn create_genesis<T: Client>(from_key: &Keypair, client: &T, amount: u64) -> Keypair {
+    let libra_genesis_key = Keypair::new();
+
+    let instruction = system_instruction::create_account(
+        &from_key.pubkey(),
+        &libra_genesis_key.pubkey(),
+        1,
+        2048 as u64,
```
Comment (sakridge) [inline programs/librapay_api/src/lib.rs]: Does it need a cast?

File: programs/librapay_api/src/librapay_transaction.rs [hunk 3]
```diff
@@ -146,18 +148,12 @@ mod tests {
...
-        let libra_mint_keypair = Keypair::new();
         let genesis_block = GenesisBlock::new(
-            &[
-                (
-                    mint_keypair.pubkey(),
-                    Account::new(lamports, 0, &system_program::id()),
-                ),
-                (libra_mint_keypair.pubkey(), libra_genesis_account),
-            ],
+            &[(
```
Comment (sakridge) [inline programs/librapay_api/src/librapay_transaction.rs]: Could use the standard genesis create helper here now.

File: programs/librapay_api/src/lib.rs (hunk 2 above)
Comment (jackcmay) [reply programs/librapay_api/src/lib.rs]: Switched it to actually calculate the size needed, which then does need the cast :-)

**Pull request merged**